addons/product/models/product.py

```diff
@@ -279,7 +279,16 @@ def _set_image_value(self, value):
         if isinstance(value, pycompat.text_type):
             value = value.encode('ascii')
         image = tools.image_resize_image_big(value)
-        if self.product_tmpl_id.image:
+
+        # This is needed because when there is only one variant, the user
+        # doesn't know there is a difference between template and variant, he
+        # expects both images to be the same.
+        variant_count = self.env['product.product'].search_count([
+            ('product_tmpl_id', '=', self.product_tmpl_id.id),
+            ('active', '=', True),
```

Comment:
if variant not activated, no archived, so not need to check the active

Reply:
Actually there is a field for that: product_variant_count so I'll use it instead.